java/bulk-export/bulk-export-lambda/src/main/java/sleeper/bulkexport/lambda/SqsBulkExportProcessorLambda.java

```diff
@@ -0,0 +1,146 @@
+    private BulkExportQuerySerDe bulkExportQuerySerDe;
+
+    /**
+     * Constructs an instance of SqsBulkExportProcessorLambda using default clients
+     * for Amazon SQS, Amazon S3, and Amazon DynamoDB.
+     *
+     * @throws ObjectFactoryException if there is an error creating the object.
+     */
+    public SqsBulkExportProcessorLambda() throws ObjectFactoryException {
+        this(AmazonSQSClientBuilder.defaultClient(), AmazonS3ClientBuilder.defaultClient(),
+                AmazonDynamoDBClientBuilder.defaultClient());
+    }
+
+    /**
+     * Constructs an instance of SqsBulkExportProcessorLambda.
+     *
+     * @param sqsClient    The Amazon SQS client used for interacting with SQS.
+     * @param s3Client     The Amazon S3 client used for interacting with S3.
+     * @param dynamoClient The Amazon DynamoDB client used for interacting with
+     *                     DynamoDB.
+     * @throws ObjectFactoryException If there is an error creating the object.
+     */
+    public SqsBulkExportProcessorLambda(AmazonSQS sqsClient, AmazonS3 s3Client, AmazonDynamoDB dynamoClient)
+            throws ObjectFactoryException {
+        this.sqsClient = sqsClient;
+        this.s3Client = s3Client;
+        this.dynamoClient = dynamoClient;
+
+        String bucket = System.getenv(CONFIG_BUCKET.toEnvironmentVariable());
+        instanceProperties = S3InstanceProperties.loadFromBucket(s3Client, bucket);
+        TablePropertiesProvider tablePropertiesProvider = S3TableProperties.createProvider(instanceProperties, s3Client,
+                dynamoClient);
+        bulkExportQuerySerDe = new BulkExportQuerySerDe();
+        processor = SqsBulkExportProcessor.builder()
+                .sqsClient(sqsClient)
+                .s3Client(s3Client)
+                .dynamoClient(dynamoClient)
+                .instanceProperties(instanceProperties)
+                .tablePropertiesProvider(tablePropertiesProvider)
+                .build();
+    }
+
+    /**
+     * Handles the incoming SQS event, processes each message, and performs the bulk
+     * export operation.
+     *
+     * @param event   The SQS event containing the messages to be processed.
+     * @param context The AWS Lambda context object providing runtime information.
+     * @return Always returns null.
+     */
+    @Override
+    public Void handleRequest(SQSEvent event, Context context) {
+        event.getRecords().stream()
+                .map(SQSEvent.SQSMessage::getBody)
+                .peek(body -> LOGGER.debug("Received message with body {}", body))
+                .flatMap(body -> deserialiseAndValidate(body).stream())
+                .peek(json -> LOGGER.debug(json.toString()))
+                .forEach(query -> {
+                    try {
+                        processor.processExport(query);
+                    } catch (ObjectFactoryException e) {
+                        LOGGER.error("Failed to process export query", e);
```

Comment:
I'm not sure we want to discard this failure? We could throw the exception all the way out, or report just the ones that failed with SQSBatchResponse. Either way would retry the messages or send them to the dead letter queue, whereas as it is these failures are just ignored.

Reply:
I can also see sending this directly to the dead letter queue, though there are some cases where this exception is retryable. It might be best to assume we need to retry this, and maybe consider splitting up the exception.